commit 2628afbffebfb0188936e02ddda7b8c9023e4750
Author: Matthias Reichl <hias@horus.com>
Date:   Sun Nov 18 15:32:28 2018 +0100

    rpi-wm8804-soundcard: configure wm8804 clocks only on rate change
    
    This should avoid clicks when stopping and immediately afterwards
    starting a stream with the same samplerate as before.
    
    Signed-off-by: Matthias Reichl <hias@horus.com>

diff --git a/sound/soc/bcm/rpi-wm8804-soundcard.c b/sound/soc/bcm/rpi-wm8804-soundcard.c
index ad42f5d35067..c1bed91e2cda 100644
--- a/sound/soc/bcm/rpi-wm8804-soundcard.c
+++ b/sound/soc/bcm/rpi-wm8804-soundcard.c
@@ -64,6 +64,7 @@ struct snd_rpi_wm8804_drvdata {
 
 static struct gpio_desc *snd_clk44gpio;
 static struct gpio_desc *snd_clk48gpio;
+static int wm8804_samplerate = 0;
 
 #define CLK_44EN_RATE 22579200UL
 #define CLK_48EN_RATE 24576000UL
@@ -117,6 +118,12 @@ static int snd_rpi_wm8804_hw_params(struct snd_pcm_substream *substream,
 	struct wm8804_clk_cfg clk_cfg;
 	int samplerate = params_rate(params);
 
+	if (samplerate == wm8804_samplerate)
+		return 0;
+
+	/* clear until all clocks are setup properly */
+	wm8804_samplerate = 0;
+
 	snd_rpi_wm8804_clk_cfg(samplerate, &clk_cfg);
 
 	pr_debug("%s samplerate: %d mclk_freq: %u mclk_div: %u sysclk: %u\n",
@@ -163,6 +170,8 @@ static int snd_rpi_wm8804_hw_params(struct snd_pcm_substream *substream,
 		return ret;
 	}
 
+	wm8804_samplerate = samplerate;
+
 	/* set sampling frequency status bits */
 	snd_soc_component_update_bits(component, WM8804_SPDTX4, 0x0f,
 			sampling_freq);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1682_2628afbffebfb0188936e02ddda7b8c9023e4750.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1682_2628afbffebfb0188936e02ddda7b8c9023e4750.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
